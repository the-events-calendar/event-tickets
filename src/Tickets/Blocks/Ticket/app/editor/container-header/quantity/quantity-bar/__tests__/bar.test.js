/**
 * External dependencies
 */
import React from 'react';
import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import Bar from './../bar';

describe( '<Bar />', () => {
	test( 'render null on default values', () => {
		const component = renderer.create( <Bar /> );
		expect( component.toJSON() ).toBe( null );
	} );

	test( 'render null on zero values', () => {
		const component = renderer.create( <Bar total={ 0 } value={ 0 } /> );
		expect( component.toJSON() ).toBe( null );
	} );

	test( 'renders 100 when percentage is above 100', () => {
		const component = shallow( <Bar value={ 200 } total={ 100 } /> );
		const span = component.get( 0 );
		const { style } = span.props;
		expect( style ).toEqual( { width: '100.00%' } );
	} );

	test( 'renders 0 when percentage is negative', () => {
		const component = shallow( <Bar value={ -200 } total={ 100 } /> );
		const span = component.get( 0 );
		const { style } = span.props;
		expect( style ).toEqual( { width: '0.00%' } );
	} );

	test( 'render percentage as inline style', () => {
		const component = shallow( <Bar value={ 33 } total={ 100 } /> );
		const span = component.get( 0 );
		const { style } = span.props;
		expect( style ).toEqual( { width: '33.00%' } );
	} );

	test( 'render percentage with custom class name', () => {
		const component = renderer.create( <Bar value={ 45 } total={ 100 } className="jest-test" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'render percentage with children', () => {
		const component = renderer.create( <Bar value={ 45 } total={ 100 }>test</Bar> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
