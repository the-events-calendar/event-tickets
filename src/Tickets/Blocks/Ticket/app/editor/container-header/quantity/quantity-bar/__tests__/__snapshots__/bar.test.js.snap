// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar /> render percentage with children 1`] = `
<span
  className="tribe-editor__quantity-bar__bar"
  style={
    Object {
      "width": "45.00%",
    }
  }
>
  test
</span>
`;

exports[`<Bar /> render percentage with custom class name 1`] = `
<span
  className="tribe-editor__quantity-bar__bar jest-test"
  style={
    Object {
      "width": "45.00%",
    }
  }
/>
`;
